### PR TITLE
Let's allow hf_hub < 1.0

### PR DESCRIPTION
**Pull request by ArthurZucker**
An attempt to be more permissive with HF hub version

### CODE REVIEW

**Comment by winglian** [comment]
@ArthurZucker when will this be merged? It's painful having to upgrade and downgrade back and forth to train and then upload  models

File: bindings/python/pyproject.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ dynamic = [
     'license',
     'readme',
 ]
-dependencies = ["huggingface_hub>=0.16.4,<0.17"]
+dependencies = ["huggingface_hub>=0.16.4,<1.0"]
```
Comment (Narsil) [inline bindings/python/pyproject.toml:29]: No, we need to allow for breaking changes in huggingface_hub meaning we can do `>=0.17,<0.18` instead.

Comment (ArthurZucker) [inline bindings/python/pyproject.toml:29]:
```suggestion
dependencies = ["huggingface_hub>=0.16.4,<0.18"]
```

**Review by Narsil — APPROVED** [review]
LGTM

**Pull request merged**